vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_authorization.cc

```diff
@@ -229,12 +225,96 @@ void UpholdAuthorization::OnCardCreate(
             address.substr(0, 5));
   }
 
-  base::flat_map<std::string, std::string> args;
   if (wallet_ptr->status != type::WalletStatus::VERIFIED) {
+    base::flat_map<std::string, std::string> args{};
     args["redirect_url"] = GetSecondStepVerify();
+    callback(type::Result::LEDGER_OK, args);
   }
 
-  callback(type::Result::LEDGER_OK, args);
+  GetAnonFunds(
+      std::bind(&UpholdAuthorization::OnGetAnonFunds, this, _1, _2, callback));
+}
+
+void UpholdAuthorization::GetAnonFunds(
+    endpoint::promotion::GetWalletBalanceCallback callback) const {
+  // if we don't have user funds in anon card anymore
+  // we can skip balance server ping
+  if (!ledger_->state()->GetFetchOldBalanceEnabled()) {
+    callback(type::Result::LEDGER_OK, type::Balance::New());
+    return;
+  }
+
+  const auto wallet = ledger_->wallet()->GetWallet();
+  if (!wallet) {
+    BLOG(1, "Wallet is not created!");
+    ledger_->state()->SetFetchOldBalanceEnabled(false);
+    callback(type::Result::LEDGER_OK, type::Balance::New());
+    return;
+  }
+
+  if (wallet->payment_id.empty()) {
+    BLOG(0, "Payment ID is empty!");
+    callback(type::Result::LEDGER_ERROR, nullptr);
+    return;
+  }
+
+  promotion_server_->get_wallet_balance()->Request(callback);
+}
+
+void UpholdAuthorization::OnGetAnonFunds(
+    const type::Result result,
+    type::BalancePtr balance,
+    ledger::ExternalWalletAuthorizationCallback callback) const {
+  if (result != type::Result::LEDGER_OK || !balance) {
+    BLOG(0, "Couldn't get anonymous funds!");
+    callback(type::Result::LEDGER_ERROR, {});
+    return;
+  }
+
+  if (balance->user_funds == 0.0) {  // == floating-point comparison!
+    ledger_->state()->SetFetchOldBalanceEnabled(false);
+  }
+
+  TransferAnonFunds(
+      balance->user_funds,
+      std::bind(&UpholdAuthorization::OnTransferAnonFunds, this, _1, callback));
+}
+
+void UpholdAuthorization::TransferAnonFunds(
```

Comment:
The primary purpose of this call is to link the user's rewards wallet to their Uphold account (transferring anonymous funds is a secondary, legacy use case), so it might make sense to name this something along the lines of `LinkWallet`.

Reply:
Resolved in 088ad1c7ccdaa1ed422fbc903352a0a5f236ef6b.